packages/providers/src/provider.ts

```diff
@@ -190,6 +190,13 @@ export type BuildPredicateOptions = {
   fundTransaction?: boolean;
 } & Pick<TransactionRequestLike, 'gasLimit' | 'gasPrice' | 'maturity'>;
 
+/*
+ * Provider initialization options
+ */
+export type ProviderOptions = {
+  fetch?: unknown;
```

Comment:
Marked `fetch`'s type as `unknown` because even the internal type for `fetch` inside `GraphQLClient`'s type declarations is `any`.

Reply:
Could we add some common typings defined by us, or would it restrict the options?

Their [docs](https://github.com/jasonkuhrt/graphql-request#using-a-custom-fetch-method) _(somehow also vague)_ cites [this](https://github.com/valeriangalliat/fetch-cookie/blob/40eb95500ee2c7d9db27e438448742b8c8ce2fbd/src/index.ts#L236-L240C32) as an example, not sure if we can draw inspiration from it.

@luizstacio What do you think?

Reply:
Typing the fetch method would be tricky as the API goes from simple type to extensive returns types, steam, JSON, etc... As this Option is more advanced and not for common use, I don't see the need to invest too much time on this fetch type.

The basic implementation I would suggest is to type the parameters that are passed to the fetch.

```ts
interface RequestOptions {
      method: 'POST',
      headers: { [key: string]: string },
      body: string
}

fetch: (url: string, options: RequestOptions) => Promise<any>
```